api/types/types.proto

```diff
@@ -1694,4 +1694,73 @@ message ResourceWithSecretsRequest {
+}
+
+// OIDCConnectorSpecV2 is an OIDC connector specification.
+//
+// It specifies configuration for Open ID Connect compatible external
+// identity provider: https://openid.net/specs/openid-connect-core-1_0.html
+message OIDCConnectorSpecV2 {
+    // Issuer URL is the endpoint of the provider, e.g. https://accounts.google.com.
+    string IssuerURL = 1 [ (gogoproto.jsontag) = "issuer_url" ];
+    // ClientID is the id of the authentication client (Teleport Auth server).
+    string ClientID = 2 [ (gogoproto.jsontag) = "client_id" ];
+    // ClientSecret is used to authenticate the client.
+    string ClientSecret = 3 [ (gogoproto.jsontag) = "client_secret" ];
+    // RedirectURL is a URL that will redirect the client's browser
+    // back to the identity provider after successful authentication.
+    // This should match the URL on the Provider's side.
+    string RedirectURL = 4 [ (gogoproto.jsontag) = "redirect_url" ];
+    // ACR is an Authentication Context Class Reference value. The meaning of the ACR
+    // value is context-specific and varies for identity providers.
+    string ACR = 5 [ (gogoproto.jsontag) = "acr_values,omitempty" ];
+    // Provider is the external identity provider.
+    string Provider = 6 [ (gogoproto.jsontag) = "provider,omitempty" ];
+    // Display is the friendly name for this provider.
+    string Display = 7 [ (gogoproto.jsontag) = "display,omitempty" ];
+    // Scope specifies additional scopes set by provider.
+    repeated string Scope = 8 [ (gogoproto.jsontag) = "scope,omitempty" ];
+    // Prompt is an optional OIDC prompt. An empty string omits prompt.
+    // If not specified, it defaults to select_account for backwards compatibility.
+    string Prompt = 9 [ (gogoproto.jsontag) = "prompt,omitempty" ];
+    // ClaimsToRoles specifies a dynamic mapping from claims to roles.
+    repeated ClaimMapping ClaimsToRoles = 10
+        [ (gogoproto.nullable) = false, (gogoproto.jsontag) = "claims_to_roles,omitempty" ];
+    // GoogleServiceAccountURI is a path to a google service account uri.
+    string GoogleServiceAccountURI = 11
+        [ (gogoproto.jsontag) = "google_service_account_uri,omitempty" ];
+    // GoogleServiceAccount is a string containing google service account credentials.
+    string GoogleServiceAccount = 12 [ (gogoproto.jsontag) = "google_service_account,omitempty" ];
+    // GoogleAdminEmail is the email of a google admin to impersonate.
+    string GoogleAdminEmail = 13 [ (gogoproto.jsontag) = "google_admin_email,omitempty" ];
+}
+
+// ClaimMapping maps an OIDC claim name to teleport roles.
+message ClaimMapping {
+    // Claim is an OIDC claim name.
+    string Claim = 1 [ (gogoproto.jsontag) = "claim" ];
+    // Value is a claim value to match.
+    string Value = 2 [ (gogoproto.jsontag) = "value" ];
+    // Roles is a list of static teleport roles to match.
+    repeated string Roles = 3 [ (gogoproto.jsontag) = "roles,omitempty" ];
```

Comment:
Nit: The `AccessRequestClaimMapping` was only previously distinct because it needed to be a protobuf type.  Its JSON representation is identical to the claim mapping, so it should be fine to just rename that type to `ClaimMapping` rather than creating a new identical protobuf type.

Reply:
@fspmarshall Good catch. Could this affect backwards compatibility though? Maybe we should replace `AccessRequestClaimMapping` in 8.0 instead.

Reply:
@Joerger How about adding a type alias for the old name with a `DELETE IN` directive?  That will keep compatibility if anyone is importing the type, but make our work simpler in the long run.